cpp/ql/src/semmle/code/cpp/ir/implementation/aliased_ssa/Operand.qll

```diff
@@ -11,13 +11,19 @@ cached
 private newtype TOperand =
   TRegisterOperand(Instruction useInstr, RegisterOperandTag tag, Instruction defInstr) {
     defInstr = Construction::getRegisterOperandDefinition(useInstr, tag) and
-    not Construction::isInCycle(useInstr)
+    not Construction::isInCycle(useInstr) and
+    strictcount(Construction::getRegisterOperandDefinition(useInstr, tag)) = 1
```

Comment:
Are the cases this is intended to filter out where the operand is a variable access (with more than one type)?

Reply:
As far as I understand, that is not the intention of these cases. [Rather, they filter out malformed IR caused by a bad snapshot or bugs in the IR construction](https://github.slack.com/archives/CP0LHP150/p1581323404337800).

Reply:
Yes, this PR addresses two issues that are probably independent: variables with multiple types and instructions with more than one of the same operand. Neither fix on its own was enough to fix value numbering on ffmpeg.

Reply:
I see, thanks both of you for clearing this up.